iOS/Flows/Home/_New/new_ConversationViewController.swift

```diff
@@ -150,36 +191,68 @@ extension new_ConversationViewController {
                                     collectionView: self.collectionView,
                                     animationCycle: animationCycle)
     }
-}
 
-extension new_ConversationViewController: ChatChannelControllerDelegate {
+    func subscribeToConversationUpdates() {
+        self.conversationController.messagesChangesPublisher.mainSink { [unowned self] changes in
+            Task {
+                await self.updateMessages(with: changes)
+            }
+        }.store(in: &self.cancellables)
+    }
+
+    func updateMessages(with changes: [ListChange<ChatMessage>]) async {
 
-    /// The controller observed changes in the `Messages` of the observed channel.
-    func channelController(_ channelController: ChatChannelController,
-                           didUpdateMessages changes: [ListChange<ChatMessage>]) {
+        guard let conversationController = self.conversationController else { return }
 
         var snapshot = self.dataSource.snapshot()
 
+        // If there's more than one change, reload all of the data.
+        guard changes.count == 1 else {
+            snapshot.deleteItems(snapshot.itemIdentifiers(inSection: .basic(conversation.cid)))
+            snapshot.appendItems(conversationController.messages.asConversationCollectionItems,
+                                 toSection: .basic(conversation.cid))
+            await self.dataSource.applySnapshotKeepingVisualOffset(snapshot,
+                                                                   collectionView: self.collectionView)
+            return
+        }
```

Comment:
If there's more than one change happening at a time, just refresh all the data. It's too difficult to handle all the deltas individually.